examples/hal_lib/makefile-common.mk

```diff
@@ -0,0 +1,132 @@
+SDCARD_MOUNT_PATH ?= /Volumes/BAREAPP
+
+LINKSCR ?= linkscript.ld
+EXTLIBDIR ?= ../../third-party
+UBOOTDIR ?= $(EXTLIBDIR)/u-boot/build
+BUILDDIR ?= build
+BINARYNAME ?= main
+UIMAGENAME ?= $(BUILDDIR)/a7-main.uimg
+
+OBJDIR = $(BUILDDIR)/obj
```

Comment:
The `shared/makefile-common.mk` has this here:
```
OBJDIR = $(BUILDDIR)/obj/obj
```

Which is an ugly hack to allow source file paths that start with `../../` to generate object files that still live in the build/ dir. (They get translated into `build/obj/obj/../../` which is `build/`, rather than `build/obj/../../` which is outside of `build/`)

I should address this probably by filtering out leading `../` before creating the OBJECTS list (line 14).